### DISCUSSION TITLE
Giving up: 11 sectors missing

### DISCUSSION THREAD

**Discussion by danielallsopp** [keirf/greaseweazle#249] [Q&A] [answered]
Hi. I've just received my Greaseweazel v4 today and connected up an old PC drive with a typical twisted cable and set about trying to dump a disk to an ADF file using the following:

`gw read dpaint.adf`

However, instead of an ADF file being created I get the following:

`Reading c=0-79:h=0-1 revs=1.1`
`Format amiga.amigados`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (46702 flux in 219.87ms)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (201475 flux in 966.96ms) (Retry #1.1)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (347298 flux in 1669.09ms) (Retry #1.2)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (494063 flux in 2376.17ms) (Retry #1.3)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (640578 flux in 3081.89ms) (Retry #2.1)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (786798 flux in 3786.13ms) (Retry #2.2)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (932756 flux in 4489.17ms) (Retry #2.3)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (1078141 flux in 5189.12ms) (Retry #3.1)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (1222756 flux in 5885.30ms) (Retry #3.2)`
`T0.0: AmigaDOS (0/11 sectors) from Raw Flux (1367107 flux in 6579.94ms) (Retry #3.3)`
`T0.0: Giving up: 11 sectors missing`

I've tried three different disks now, all of which read fine on a real Amiga 4000 and show up on Workbench. They're not copy protected disks: one is a Workbench 3.1 disk, the other Fonts 3.1 and the other Deluxe Paint 3, so standard AmigaDOS disk formats.

If I try and write an ADF file back to a disk then the process will complete as expected, but the disk is not readable with the real Amiga nor with FS-UAE.

I'm running the latest GW software on macOS Monterey and the GW reports the following with the `info` command:

`Host Tools: 1.3`
`Device:`
`  Port:     /dev/cu.usbmodem1464101`
`  Model:    Greaseweazle V4`
`  MCU:      AT32F403A, 144MHz, 96kB SRAM`
`  Firmware: 1.2`
`  Serial:   GW00B542624C874000079D8716`
`  USB Rate: Full Speed (12 Mbit/s)`

I have no idea of the make of the drive but it says it's made in Malaysia and has ALPS ELECTRIC CO LTD on the label.

Anyone got any ideas? I've done a search and can't find anyone with the same issues.

**Comment by keirf** [comment, marked as answer]
Your drive is probably out of alignment.

**Reply by danielallsopp** [reply]
Thanks for the reply. What's the best way to remedy that? Bin it and buy another one and hope it's not the same?

**Reply by keirf** [reply]
You can attempt realignment but it's simpler to buy another drive or small job lot of drives.

**Reply by danielallsopp** [reply]
I've acquired another drive and everything works as normal now. Thanks a lot for the hint :)

**Comment by vonj** [comment]
I have tried two drives, one Sony MPF920 Z, one NEC FD1231H.

The NEC:
gw seek 0   **works**
gw seek 1
 **FATAL ERROR:
Track0 signal asserted after seek to cylinder 1**
Seek to any other cylinder gives the same error (except cylinder number)

The Sony is the opposite.
Seeking to cylinder 1-83 is OK.
Seeking to cylinder 0:
**FATAL ERROR:
Track0 signal absent after seek to cylinder 0**

If I patch the python code (lines 268-270 in usb.py) to ignore these errors, neither drive can read anything:

**T0.0: Giving up: 18 sectors missing**

this repeats for the whole floppy.


I have tried several (PC 1440k) floppies of various makes and vintage, all with data on them, same result.

The NEC drive came with a cable with the little twist, but it also had one of the holes plugged, so it won't fit in weazle pinout port.

The Sony came with a straight cable, which I have used for both drives.

Flipping the cable makes the drive shine green LED light all the time and the weazle not detecting the drive.

After mucking about for awhile, the Sony says:
gw seek 1
**FATAL ERROR:
Track0 signal asserted after seek to cylinder 1**

gw seek 2
gw seek 0
gw seek 83
these are okay

**Reply by keirf** [reply]
The drives aren't seeking properly. They need servicing (eg wormgear clean and lube) and may also need calibration.

**Reply by vonj** [reply]
I put an old PC together and tested the drives in them, both the NEC and the Sony work perfectly in the old PC.

**Reply by keirf** [reply]
Are you specifying `--drive=b` on the command line since you are using a straight cable?

**Reply by vonj** [reply]
It only accepts either no drive specified or **--drive=a**  

Instead I changed to a twisted cable (I pulled the plastic plug from pin hole 3 in the cable which came with the NEC), but same results.

**Reply by keirf** [reply]
What sort of Greaseweazle device is it?

**Reply by keirf** [reply]
To answer myself, it must be an F1 unit. Is it based on an F1 adapter PCB, or are the drives being directly wired to the Blue Pill?

**Reply by keirf** [reply]
Is the unit updated to latest firmware (check `gw info`)?

**Comment by vonj** [comment]
Amiga-DrawBridge/Greaseweazle at main · tebl/Amiga-DrawBridgegithub.comUsing the upgrade command said it was already the latest version, 1.3 I think 12 mars 2023 kl. 15:39 skrev Keir Fraser ***@***.***>:﻿
Is the unit updated to latest firmware (check gw info)?

**Reply by keirf** [reply]
Probably some hardware issue on the Greaseweazle side, I guess. Or maybe cabling? Power? Various hardware setup issues it could be.
